### PR TITLE
Bug 2086972: test: adds test cases to image builder and describe pkgs and add go sec fixes

### PR DESCRIPTION
**Pull request by jpower432**
# Description

Adds improvements and additional test cases to Run method on
ImageBuilder.

Comment cleanup and applies fixes from staticcheck results

Adds unit test for describe Run function

Adds gosec fixes

## Type of change

Please delete options that are not relevant.

- [X] Adds tests improvements

# How Has This Been Tested?

- [X] Adds tests

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/cli/mirror/cincinnati_graph_image.go [hunk 1]
```diff
@@ -128,7 +125,7 @@ func (o *MirrorOptions) buildGraphImage(ctx context.Context, dstDir string) (ima
 }
 
 // downloadsGraphData will download the current Cincinnati graph data
-func downloadGraphData(ctx context.Context, dir, url string) error {
+func downloadGraphData(ctx context.Context, dir string) error {
```
Comment (dinhxuanvu) [inline pkg/cli/mirror/cincinnati_graph_image.go:128]: The `dir` input should be sanitized before use given it is a directory path which can contain plenty of random stuff. This was one of the few things I fixed in my prodsec PR. We should continue to keep the practice going.

**Review by dinhxuanvu — COMMENTED** [review]
Looking good. Just one nit.

File: pkg/cli/mirror/cincinnati_graph_image.go (hunk 1 above)
Comment (jpower432) [reply pkg/cli/mirror/cincinnati_graph_image.go:128]: Good feedback. This is actually getting sanitized before it is opened/used. Line number below.

https://github.com/openshift/oc-mirror/blob/7bde4c7bde6b829a34592b2d12115fb38b15576f/pkg/cli/mirror/cincinnati_graph_image.go#L136

I did a go sec run through and made a few fixes though.

**Review by dinhxuanvu — COMMENTED** [review]
/lgtm

**Pull request merged**